src/axom/core/memory_management.hpp

```diff
@@ -200,23 +194,47 @@ inline void deallocate( T*& pointer ) noexcept
 template < typename T >
 inline T* reallocate( T* pointer, std::size_t n ) noexcept
 {
-  if ( n == 0 )
-  {
-    axom::deallocate( pointer );
-    return nullptr;
-  }
-
   const std::size_t numbytes = n * sizeof( T );
 
 #ifdef AXOM_USE_UMPIRE
 
+  // Workaround for bug in Umpire's handling on reallocate(0)
```

Comment:
Note: These workarounds will be necessary until we and all our customers build against an as-yet-unreleased version of Umpire (post v1.1).

Should we guard these with a config option? E.g. ``AXOM_USE_UMPIRE_REALLOC_WORKAROUND`` which will be ``ON`` by default until we upgrade Umpire and then ``OFF`` by default?

Reply:
Since it appears that an Umpire release is imminent (they started the release process yesterday, I believe), I say leave this as is and clean it up when we pull in the new Umpire.